### ISSUE TITLE
Request: OpenSmartHouse device parameter change for Fibaro Walli Dimmer

### ISSUE THREAD

**Issue by jarlesmartthings** [jvmahon/HubitatCustom#10]
There are some missing options under parameter 40 and 41 for the Fibaro Walli Dimmer.

Device: https://www.opensmarthouse.org/zwavedatabase/1074/

The options below should be available for both parameter **40** (First button – scenes sent) and **41** (Second button – scenes sent). Currently 41 is free text, but it should have the same options as parameter 40 as well.

Looking at the device link above, parameter 40 there's a red symbol behind the parameter description. Does that actually mean multiple choice / bitmask or something? Because it feels kind of weird to add all of these combinations if there's already support for combinations of values in the database.

If we need to add these, we would have to do something like this to cover all combinations:

Value | Name | Currently missing
-- | -- | --
1 | Key pressed once | NO
2 | Key pressed twice | NO
3 | Key pressed once or twice | YES
4 | Key pressed 3 times | NO
5 | Key pressed once or 3 times | YES
6 | Key pressed twice or 3 times | YES
7 | Key pressed once, twice or 3 times | YES
8 | Key held down and released | NO
9 | Key pressed once or held / released | YES
10 | Key pressed twice or held / released | YES
11 | Key pressed once, twice or held / released | YES
12 | Key pressed 3 times or held / released | YES
13 | Key pressed once, 3 times or held / released | YES
14 | Key pressed twice, 3 times or held / released | YES
15 | All actions | YES

Device parameter documentation can be found on page 28 in the manual: https://manuals.fibaro.com/content/manuals/en/FGWDEU-111/FGWDEU-111-T-EN-v1.0.pdf

**Comment by jvmahon** [comment]
Parameter 40 was incorrectly entered in the database as an enumerated data type allowing selection of only 1 of the options. I've changed it to be like parameter 41 where you'll have to enter the proper number which you'll have to figure yourself (i.e., add together the various option values; use 15 for them all). Hubitat doesn't have an easy way to input bitmask, so this generally works a bit better. It may take a few days for the moderator to approve. Give it a day or two, then click "Reset Driver State Data" followed by "Initialize". If its working, close out this issue. If it isn't fixed within a few days, post a reminder and I'll check the database again.

**Comment by jarlesmartthings** [comment]
Great! That's probably a better solution if Hubitat doesn't support this yet. 👍

**Comment by jarlesmartthings** [comment]
It seems like last approval was 2021-01-16 09:19:23 (and the red icon behind parameter 40 is gone), but the API still returns parameter 40 with `limit_options: 1`

Is that the property you changed, @jvmahon? Is so, I'll give it a few more days 👍

**Comment by jvmahon** [comment]
Yep, I missed that when editing. I just did another update.
Give it another day, then click on "Reset Driver State Data" to remove the old parameter input information, then finally, select "Initialize"   (or, if you hadn't done this reset sequence before, just try it now -- I don't think the "limit_options:1" should have had any effect once I got rid of the other options).

**Comment by jarlesmartthings** [comment]
So I'm not really sure what happened here. I've tried to click on "Reset Driver State Data" and then "Initialize" several times, both today and earlier, but parameter 40 always sticked to"enum". 

Now, I went into your code and hardcoded to `newInput.type = "number"`, to verify that the code executed at all. When I then clicked on "Reset Driver State Data" followed by "Initialize" and a page refresh, all parameters disappeared. After a while, a new page refresh suddenly displayed parameter 40 as number (I guess the delay here is expected).

But after this I changed it back to enum (if there's options) and after reset and initialize it was still number. Then a went over to another device and did  "Reset Driver State Data" followed by "Initialize", and it immediately changed to number. So, I'm not sure what happened here, but it seems to work now :-)

(And I agree, looking at the code, there's no reason "limit_options" should have anything to do with it.)
